plugins/woocommerce/tests/e2e-pw/tests/merchant/create-variable-product.spec.js

```diff
@@ -34,206 +37,678 @@ async function deleteProductsAddedByTests( baseURL ) {
+			async () => {
+				await expect(
+					page.getByText( 'Product draft updated. ' )
+				).toBeVisible();
+			}
+		);
 
-		await page.getByRole( 'button', { name: 'Save attributes'} ).click( { clickCount: 3 });
+		await test.step( 'Click on the "Variations" tab.', async () => {
+			await page.click( 'a[href="#variable_product_options"]' );
+		} );
 
-		// wait for the tour's dismissal to be saved
-		await page.waitForResponse(
-			( response ) =>
-				response.url().includes( '/post.php' ) &&
-				response.status() === 200
+		await test.step(
+			'Click on the "Generate variations" button.',
+			async () => {
+				// event listener for handling the link_all_variations confirmation dialog
+				page.on( 'dialog', ( dialog ) => dialog.accept() );
+
+				await page.click( 'button.generate_variations' );
+			}
 		);
 
-		// Save before going to the Variations tab to prevent variations from all attributes to be automatically created
-		await page.locator( '#save-post' ).click();
-		await expect(
-			page.getByText( 'Product draft updated. ' )
-		).toBeVisible();
+		await test.step(
+			'Expect variations to have the correct attribute values',
+			async () => {
+				for ( let i = 0; i < 8; i++ ) {
+					const val1 = 'val1';
+					const val2 = 'val2';
+					const attr3 = !! ( i % 2 ); // 0-1,4-5 / 2-3,6-7
+					const attr2 = i % 4 > 1; // 0-3 / 4-7
+					const attr1 = i > 3;
+					await expect(
+						page.locator(
+							`select[name="attribute_attr-1[${ i }]"]`
+						)
+					).toHaveValue( attr1 ? val2 : val1 );
+					await expect(
+						page.locator(
+							`select[name="attribute_attr-2[${ i }]"]`
+						)
+					).toHaveValue( attr2 ? val2 : val1 );
+					await expect(
+						page.locator(
+							`select[name="attribute_attr-3[${ i }]"]`
+						)
+					).toHaveValue( attr3 ? val2 : val1 );
+				}
+			}
+		);
 
-		await page.click( 'a[href="#variable_product_options"]' );
+		await test.step( 'Click "Save Draft" button.', async () => {
+			await page.locator( '#save-post' ).click();
+		} );
 
-		// event listener for handling the link_all_variations confirmation dialog
-		page.on( 'dialog', ( dialog ) => dialog.accept() );
+		await test.step(
+			'Expect the "Product draft updated." notice to appear.',
+			async () => {
+				await expect(
+					page.locator( '#message.notice-success' )
+				).toContainText( 'Product draft updated.' );
+			}
+		);
+	} );
 
-		// generate variations from all attributes
-		await page.click( 'button.generate_variations' );
+	test( 'can individually edit variations', async ( { page } ) => {
+		const variationRows = page.locator( '.woocommerce_variation' );
+		const firstVariation = variationRows.filter( {
+			hasText: `#${ fixedVariationIds[ 0 ] }`,
+		} );
+		const secondVariation = variationRows.filter( {
+			hasText: `#${ fixedVariationIds[ 1 ] }`,
+		} );
+		const thirdVariation = variationRows.filter( {
+			hasText: `#${ fixedVariationIds[ 2 ] }`,
+		} );
```

Comment:
When locating the row of the variation to edit (and later on, verify), I changed the approach from relying on how they're positioned in the UI, to looking at their variation ID, which I think is a more stable way of locating them.